service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapperTest.java

```diff
@@ -149,46 +165,154 @@ public void setUp() throws IOException {
+    }
 
-        String actualXml = observationMapper.mapObservationToCompoundStatement(
-            observationAssociatedWithSpecimen
+    @Test
+    void When_MappingTestResult_With_NopatMetaSecurity_Expect_ConfidentialityCodeWithinObservationStatement()
+        throws XPathExpressionException, IOException, ParserConfigurationException, SAXException {
+        final Observation observation = getObservationResourceFromJson(OBSERVATION_TEST_RESULT_JSON);
+        final String xpath = "/component/ObservationStatement/confidentialityCode";
+
+        ConfidentialityCodeUtility.appendNopatSecurityToMetaForResource(observation);
+        when(confidentialityService.generateConfidentialityCode(observation))
+            .thenReturn(Optional.of(NOPAT_CONFIDENTIALITY_CODE));
+
+        final String actualXml = observationMapper
+            .mapObservationToCompoundStatement(observation);
+
+        assertThat(XmlParsingUtility.xpathMatchFound(actualXml, xpath)).isTrue();
+    }
+
+    @Test
+    void When_MappingTestResult_With_NoscrubMetaSecurity_Expect_ConfidentialityCodeNotPresent() {
+        final Observation observation = getObservationResourceFromJson(OBSERVATION_TEST_RESULT_JSON);
+
+        ConfidentialityCodeUtility.appendNoscrubSecurityToMetaForResource(observation);
+        when(confidentialityService.generateConfidentialityCode(observation))
+            .thenReturn(Optional.empty());
+
+        final String actualXml = observationMapper
+            .mapObservationToCompoundStatement(observation);
+
+        assertThat(actualXml).doesNotContainIgnoringCase(NOPAT_CONFIDENTIALITY_CODE);
+    }
+
+    @Test
+    void When_MappingFilingComment_With_NopatMetaSecurity_Expect_ConfidentialityCodeWithinNarrativeStatement()
+        throws XPathExpressionException, IOException, ParserConfigurationException, SAXException {
+        final Observation observation = getObservationResourceFromJson(OBSERVATION_FILING_COMMENT_JSON);
+        final String xPath = "/component/NarrativeStatement/confidentialityCode";
+
+        ConfidentialityCodeUtility.appendNopatSecurityToMetaForResource(observation);
+        when(confidentialityService.generateConfidentialityCode(observation))
+            .thenReturn(Optional.of(NOPAT_CONFIDENTIALITY_CODE));
+
+        final String actualXml = observationMapper
+            .mapObservationToCompoundStatement(observation);
+
+        assertThat(XmlParsingUtility.xpathMatchFound(actualXml, xPath)).isTrue();
+    }
+
+    @Test
+    void When_MappingFilingComment_With_NopatMetaSecurity_Expect_ConfidentialityCodeNotPresent() {
+        final Observation observation = getObservationResourceFromJson(OBSERVATION_FILING_COMMENT_JSON);
+
+        ConfidentialityCodeUtility.appendNoscrubSecurityToMetaForResource(observation);
+        when(confidentialityService.generateConfidentialityCode(observation))
+            .thenReturn(Optional.empty());
+
+        final String actualXml = observationMapper
```

Comment:
in my opinion 
final String actualXml = observationMapper.mapObservationToCompoundStatement(observation);
is perfectly OK as one-liner and it desn't violate the checkstyle rules...